src/catalua.h

```diff
@@ -3,9 +3,90 @@
 #define CATALUA_H
 
 #include "int_id.h"
+#include "enums.h"
+#include "item.h"
 
 #include <string>
 #include <sstream>
+#include <vector>
+#include <list>
+
+/**
+ * Class that contains callback argument name and argument value (in different types).
+ */
+class CallbackArgument
+{
+    private:
+        std::string name;
+        std::string type;
+        int value_integer;
+        double value_double;
+        float value_float;
+        std::string value_string;
+        tripoint value_tripoint;
+        item value_item;
+    public:
+        CallbackArgument( std::string arg_name, int arg_value ) {
+            name = arg_name;
+            type = "integer";
+            value_integer = arg_value;
+        };
+        CallbackArgument( std::string arg_name, double arg_value ) {
+            name = arg_name;
+            type = "double";
+            value_double = arg_value;
+        };
+        CallbackArgument( std::string arg_name, float arg_value ) {
+            name = arg_name;
+            type = "float";
+            value_float = arg_value;
+        };
+        CallbackArgument( std::string arg_name, std::string arg_value ) {
+            name = arg_name;
+            type = "string";
+            value_string = arg_value;
+        };
+        CallbackArgument( std::string arg_name, tripoint arg_value ) {
+            name = arg_name;
+            type = "tripoint";
+            value_tripoint = arg_value;
+        };
+        CallbackArgument( std::string arg_name, item arg_value ) {
+            name = arg_name;
+            type = "item";
+            value_item = arg_value;
+        };
+
+        std::string GetName() {
+            return name;
+        }
+        std::string GetType() {
+            return type;
+        }
+        int GetValueInt() {
+            return value_integer;
+        }
+        int GetValueDouble() {
+            return value_double;
+        }
+        int GetValueFloat() {
+            return value_float;
+        }
+        std::string GetValueString() {
+            return value_string;
+        }
+        tripoint GetValueTripoint() {
```

Comment:
All these getter functions are non-const, why? They don't change the object in any way.